pallets/xyk/src/lib.rs

```diff
@@ -470,10 +493,10 @@ decl_module! {
 
 impl<T: Trait> Module<T> {
     pub fn calculate_sell_price(
-        input_reserve: T::Balance,
-        output_reserve: T::Balance,
-        sell_amount: T::Balance,
-    ) -> T::Balance {
+        input_reserve: BalanceOf<T>,
```

Comment:
These functions also seem to be susceptible to divide by zero.
We need to check the entire pallet for divide by zero.